crates/astria-sequencer-relayer/src/serde.rs

```diff
@@ -1,4 +1,42 @@
 //! Utilities for serializing and deserializing bytes
+use std::collections::HashMap;
+
 use base64_serde::base64_serde_type;
+use serde::{
+    ser::SerializeMap,
+    Serialize,
+};
+
+use crate::types::{
+    IndexedTransaction,
+    Namespace,
+};
 
 base64_serde_type!(pub(crate) Base64Standard, base64::engine::general_purpose::STANDARD);
+
+pub(crate) struct NamespaceToTxCount<'a>(
+    pub(crate) &'a HashMap<Namespace, Vec<IndexedTransaction>>,
+);
+
+impl<'a> Serialize for NamespaceToTxCount<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.0.len()))?;
+        for (ns, txs) in self.0 {
+            map.serialize_entry(&ns, &txs.len())?;
+        }
+        map.end()
+    }
+}
+
+impl<'a> std::fmt::Display for NamespaceToTxCount<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Note that this technically throws away the error, but that's
```

Comment:
Please delete this before merging. It's absolutely wrong and I was too tired last night when I wrote that.

The actual comment should be:

This cannot fail because we are only serializing into a string (unless the system is OOM).